src/odemis/gui/comp/viewport.py

```diff
@@ -781,17 +783,108 @@ class AngularResolvedViewport(ViewPort):
 
     # Default class
     canvas_class = miccanvas.AngularResolvedCanvas
-    bottom_legend_class = None
+    bottom_legend_class = RadioLegend
+
+    def __init__(self, *args, **kwargs):
+        super(AngularResolvedViewport, self).__init__(*args, **kwargs)
+        # TODO why not sufficient to place here?
+        # bottom_legend_class = RadioLegend
 
     def setView(self, view, tab_data):
+        """
+        Set the microscope view that this viewport is displaying/representing
+        *Important*: Should be called only once, at initialisation.
+
+        :param view:(model.View)
+        :param tab_data: (model.MicroscopyGUIData)
+        """
         assert(self._view is None)
 
         self._view = view
         self._tab_data_model = tab_data
+        self.stream = None
 
         # canvas handles also directly some of the view properties
         self.canvas.setView(view, tab_data)
 
+        view.stream_tree.flat.subscribe(self.connect_stream)
+
+        self.bottom_legend.set_pol_callback(self.on_pol_change)
+
+    def connect_stream(self, _):
+        """ Find the most appropriate stream in the view to be displayed, and make sure the display
+        is updated when the stream is updated.
+
+        """
+
+        ss = self._view.getStreams()
+
+        if not ss:
+            stream = None
+        elif len(ss) > 1:
+            # => pick the first one playing
+            for s in ss:
+                if s.should_update.value:
+                    stream = s
+                    break
+            else:  # no stream playing
+                logging.warning("Found %d streams, will pick one randomly", len(ss))
+                if self.stream in ss:
+                    stream = self.stream  # don't change
+                else:
+                    stream = ss[0]
+        else:
+            stream = ss[0]
+
+        if self.stream is stream:
+            logging.debug("not reconnecting to stream as it's already connected")
+            return
+
+        # Disconnect from the old VAs
+        if self.stream:
+            logging.debug("Disconnecting %s from polarization VA", stream)
+            if hasattr(self.stream, 'polarization'):
+                self.stream.polarization.unsubscribe(self.on_legend_pol_change)
+
+        # Connect to the new VA
+        self.stream = stream
+        if stream:
+            logging.debug("Connecting %s to plotviewport", stream)
+
+        # get polarization positions and set callback function
+        if hasattr(self.stream, "polarization"):
+            self.legend_get_choices()
+            self.stream.polarization.subscribe(self.on_legend_pol_change)
+            self.bottom_legend.Show(True)
+            self.Layout()
+        else:
+            self.bottom_legend.Show(False)
```

Comment:
Add a comment that the else happens in two different cases: the stream has no `.polarization` _or_ there is no stream (`stream is None`).
I'd suggest to also call `self.bottom_legend.clear()` in such case. Not only that would make it do the same as in the other viewports, but `clear()` could also automatically drop the callback.